### PR TITLE
Add mocha test runner, a unit test and istanbul test coverage report

### PR DESCRIPTION
**Pull request by madSimonJ**
I've added in Mocha test runner using Chai assertions, along with a single unit test.  I've also added a test coverage reporter.

### CODE REVIEW

**Comment by madSimonJ** [comment]
How do I add the Hacktoberfest tag to this PR?

**Comment by lynnaloo** [comment]
@madSimonJ I do it for you before I pull it in. I'm working through the PRs by the order they go in.

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{    
+    "node" : true,
```
Comment (lynnaloo) [inline .jshintrc]: We don't need this because we're already using eslint. You can add globals there as well.

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
```
Comment (lynnaloo) [inline .babelrc]: I'd prefer it if you could do this without running babel on the server code and stick to what we get for free with Node 6. I didn't really specify that in the issues, but I'll update with some better parameters.

**Comment by lynnaloo** [comment]
@madSimonJ I added some comments for just some opinionated suggestions, but I love how far you went with this. Happy #hacktoberfest!

File: .jshintrc (hunk 1 above)
Comment (madSimonJ) [reply .jshintrc]: Removed.

File: .babelrc (hunk 2 above)
Comment (madSimonJ) [reply .babelrc]: ok, I've reverted it

**Comment by madSimonJ** [comment]
I've not come across travis CI before.  Is that a Git Hub thing, or is it something in the project?

**Comment by madSimonJ** [comment]
Finally!  All CI tests passing!

**Comment by lynnaloo** [comment]
Travis-CI is its own continuous integration build system, but they have a pretty cool Github integration that I have turned on here. That `.travis.yml` file that I put in the root tells it what kind of environment to run the code in. If you specificy `node`, then it automatically runs `npm i` and `npm test`. I added `npm run lint` so it would run the eslint rules as well.

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (lynnaloo) [inline .eslintrc]: You don't need this. There's already an `.eslintrc.json` file in the root that extends the config that I use on all my projects.

**Comment by lynnaloo** [comment]
This one is next in line once you delete the redundant `.eslintrc` file and rebase.

File: .eslintrc (hunk 3 above)
Comment (madSimonJ) [reply .eslintrc]: Removed

**Comment by madSimonJ** [comment]
I think we might be ready to go now!

**Pull request merged**